### ISSUE TITLE
MuonGun weighting failing with simulation V05 MCTree (Trac #1941)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1933]
when running muongun weighting (using metaproject simulation V05-00-06) on simulation processed with simulation V05-00-06, the following error occurs:

```text
FATAL (Tree): Assertion failed: storage_ != treePtr_->end_ (I3MCTree.h:221 in ValueType& TreeBase::Tree<T, Key, Hash>::iterator_storage_impl<boost::optional<Key>, ValueType, TreeType, TreeHashMapIter>::dereference() const [with ValueType = const I3Particle, TreeType = const TreeBase::Tree<I3Particle, I3ParticleID, __gnu_cxx::hash<I3ParticleID> >, TreeHashMapIter = __gnu_cxx::_Hashtable_const_iterator<std::pair<const I3ParticleID, TreeBase::TreeNode<I3Particle> >, I3ParticleID, __gnu_cxx::hash<I3ParticleID>, std::_Select1st<std::pair<const I3ParticleID, TreeBase::TreeNode<I3Particle> > >, std::equal_to<I3ParticleID>, std::allocator<TreeBase::TreeNode<I3Particle> > >, T = I3Particle, Key = I3ParticleID, Hash = __gnu_cxx::hash<I3ParticleID>])
ERROR (I3Module): MuonWeight_Hoerandel5: Exception thrown (I3Module.cxx:116 in void I3Module::Do(void (I3Module::*)()))
Traceback (most recent call last):
  File "./IC86_2011_L3b_muongun.py", line 182, in <module>
    tray.Execute()
  File "/cvmfs/icecube.opensciencegrid.org/py2-v2/RHEL_6_x86_64/metaprojects/simulation/V05-00-06/lib/I3Tray.py", line 234, in Execute
    super(I3Tray, self).Execute()
RuntimeError: Assertion failed: storage_ != treePtr_->end_ (in ValueType& TreeBase::Tree<T, Key, Hash>::iterator_storage_impl<boost::optional<Key>, ValueType, TreeType, TreeHashMapIter>::dereference() const [with ValueType = const I3Particle, TreeType = const TreeBase::Tree<I3Particle, I3ParticleID, __gnu_cxx::hash<I3ParticleID> >, TreeHashMapIter = __gnu_cxx::_Hashtable_const_iterator<std::pair<const I3ParticleID, TreeBase::TreeNode<I3Particle> >, I3ParticleID, __gnu_cxx::hash<I3ParticleID>, std::_Select1st<std::pair<const I3ParticleID, TreeBase::TreeNode<I3Particle> > >, std::equal_to<I3ParticleID>, std::allocator<TreeBase::TreeNode<I3Particle> > >, T = I3Particle, Key = I3ParticleID, Hash = __gnu_cxx::hash<I3ParticleID>])
```

Migrated from https://code.icecube.wisc.edu/ticket/1941
```json
{
    "status": "closed", 
    "changetime": "2017-02-02T10:55:36", 
    "description": "when running muongun weighting (using metaproject simulation V05-00-06) on simulation processed with simulation V05-00-06, the following error occurs:\n\n{{{\nFATAL (Tree): Assertion failed: storage_ != treePtr_->end_ (I3MCTree.h:221 in ValueType& TreeBase::Tree<T, Key, Hash>::iterator_storage_impl<boost::optional<Key>, ValueType, TreeType, TreeHashMapIter>::dereference() const [with ValueType = const I3Particle, TreeType = const TreeBase::Tree<I3Particle, I3ParticleID, __gnu_cxx::hash<I3ParticleID> >, TreeHashMapIter = __gnu_cxx::_Hashtable_const_iterator<std::pair<const I3ParticleID, TreeBase::TreeNode<I3Particle> >, I3ParticleID, __gnu_cxx::hash<I3ParticleID>, std::_Select1st<std::pair<const I3ParticleID, TreeBase::TreeNode<I3Particle> > >, std::equal_to<I3ParticleID>, std::allocator<TreeBase::TreeNode<I3Particle> > >, T = I3Particle, Key = I3ParticleID, Hash = __gnu_cxx::hash<I3ParticleID>])\nERROR (I3Module): MuonWeight_Hoerandel5: Exception thrown (I3Module.cxx:116 in void I3Module::Do(void (I3Module::*)()))\nTraceback (most recent call last):\n  File \"./IC86_2011_L3b_muongun.py\", line 182, in <module>\n    tray.Execute()\n  File \"/cvmfs/icecube.opensciencegrid.org/py2-v2/RHEL_6_x86_64/metaprojects/simulation/V05-00-06/lib/I3Tray.py\", line 234, in Execute\n    super(I3Tray, self).Execute()\nRuntimeError: Assertion failed: storage_ != treePtr_->end_ (in ValueType& TreeBase::Tree<T, Key, Hash>::iterator_storage_impl<boost::optional<Key>, ValueType, TreeType, TreeHashMapIter>::dereference() const [with ValueType = const I3Particle, TreeType = const TreeBase::Tree<I3Particle, I3ParticleID, __gnu_cxx::hash<I3ParticleID> >, TreeHashMapIter = __gnu_cxx::_Hashtable_const_iterator<std::pair<const I3ParticleID, TreeBase::TreeNode<I3Particle> >, I3ParticleID, __gnu_cxx::hash<I3ParticleID>, std::_Select1st<std::pair<const I3ParticleID, TreeBase::TreeNode<I3Particle> > >, std::equal_to<I3ParticleID>, std::allocator<TreeBase::TreeNode<I3Particle> > >, T = I3Particle, Key = I3ParticleID, Hash = __gnu_cxx::hash<I3ParticleID>])\n}}}", 
    "reporter": "nwandkowsky", 
    "cc": "", 
    "resolution": "duplicate", 
    "_ts": "1486032936727351", 
    "component": "cmake", 
    "summary": "MuonGun weighting failing with simulation V05 MCTree", 
    "priority": "normal", 
    "keywords": "MuonGun", 
    "time": "2017-01-25T20:18:44", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20170125T20:28:15: david.schultz commented:

> Someone is trying to dereference an invalid iterator?
> 
> Maybe converting between iterator types for the "end" of the tree.

**Comment by icecube-trac** [comment]
Trac update at 20170202T10:55:36: 
* jvansanten commented:

> Fixed just like #1575


* jvansanten changed status from "new" to "closed"
* jvansanten changed resolution from "" to "duplicate"
